frontend/packages/component-library/src/button/_BaseButton.tsx

```diff
@@ -108,8 +108,13 @@ const checkButtonPropsForErrors = ({
   color,
   type,
 }: _BaseButtonProps) => {
-  if (color === 'gray' && type !== 'secondary') {
-    throw new Error('Expect type prop to be secondary when color is gray');
+  if (
+    (color === 'gray' && type == 'primary') ||
+    (color === 'gray' && type === 'tertiary' && !isIconOnly)
+  ) {
```

Comment:
Ideally we shouldn't throw errors for invalid props, perhaps console warnings at the most and force the component into reasonable default states. Errors will crash the whole page making for a poor UX. Given this is an existing check, it's ok to update it but we should consider going back and removing errors where possible.

If we want to prevent contributors from passing invalid props (which I believe is the intention here), we can use Typescript for that using unions which will result in a compile time error rather than a runtime error.

Reply:
IIRC we had tried to get this working with type checking in the [original Button PR](https://github.com/code-dot-org/code-dot-org/pull/56925) but there were some issues. That being said, now that this has been in use and we have a firmer idea of what the props are, maybe it's worth another shot? Generally agreed that compile time checking would be better than runtime checking if possible.